api/src/services/graphql/utils/process-error.ts

```diff
@@ -1,12 +1,19 @@
-import { isDirectusError } from '@directus/errors';
+import { isDirectusError, type DirectusError } from '@directus/errors';
 import type { Accountability } from '@directus/types';
 import type { GraphQLError, GraphQLFormattedError } from 'graphql';
 import logger from '../../../logger.js';
 
-const processError = (accountability: Accountability | null, error: Readonly<GraphQLError>): GraphQLFormattedError => {
+const processError = (
+	accountability: Accountability | null,
+	error: Readonly<GraphQLError & { originalError: GraphQLError | DirectusError | Error | undefined }>,
```

Comment:
`error.originalError` is an instance of `GraphQLError` in all the cases i tried. however i have kept the `originalError` type as union to keep it consistent/backward compatible with type `GraphQLError`